### PR TITLE
User story 19 20

### PR DESCRIPTION
**Pull request by muydanny**
completed

### CODE REVIEW

File: spec/features/apps/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ end
+
+  it "view all applications for individual pet" do
+
+    visit "/pets/#{@pet1}"
+    
+  end
+end
+
+# When I visit a pets show page
```
Comment (muydanny) [inline spec/features/apps/index_spec.rb:51]: Before eval we can go through and delete all these

File: spec/features/apps/index_spec.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
    @pet222 = @shelter2.pets.create(name: "P222",age: "1",sex: "male",description: "d1",image: "i1",adoptable: true)
    @pet3 = @shelter3.pets.create(name: "P3",age: "1",sex: "male",description: "d1",image: "i1",adoptable: true)
 
-   @app1 = App.create(name:"A1", address: "a1", city: "C1", state: "ST", zip: "12345", phone_number: "12345678", description:"desc",approved:"false")
+   @app1 = App.create(name:"A1X", address: "a1", city: "C1", state: "ST", zip: "12345", phone_number: "12345678", description:"desc",approved:"false")
```
Comment (muydanny) [inline spec/features/apps/index_spec.rb:17]: A1X is our team alias

File: app/models/app.rb [hunk 3]
```diff
@@ -10,4 +10,8 @@ class App < ApplicationRecord
   validates_presence_of :zip
   validates_presence_of :description
 
+  def pets_list
+    pets.pluck(:name).join(", ")
```
Comment (perryr16) [inline app/models/app.rb:14]: nice use of pluck and join combo

File: app/views/apps/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<div class="content">
+
+  <h1 class="header">Applications for <%= @pet.name %> </h1>
+
+  <h2> Applicant Name: </h2>
+  <br>
+  <% @pet.apps.each do |app| %>
+  <section id = "app-<%= app.id %>"  >
+    <p><a class= "pet-link" href="/apps/<%= app.id %>"><%= app.name %></a></p>
```
Comment (perryr16) [inline app/views/apps/index.html.erb:9]: We are using both the <a> and the link_to tags. if we have time we should standardize. add it to the tech debt

File: spec/features/apps/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    within("#app-#{@app1.id}") do
+      expect(page).to have_content(@app1.name)
+      expect(page).to have_css(".pet-link")
+      expect(page).to_not have_content(@app11.name)
+    end
+
+    within("#app-#{@app11.id}") do
+      expect(page).to have_content(@app11.name)
+      expect(page).to_not have_content(@app1.name)
```
Comment (perryr16) [inline spec/features/apps/index_spec.rb:43]: solid test

**Pull request merged**